# Real Estate Value Trends
*(Before starting, please click the circled minus icon “⊖” on the right of the screen to see the external view with nbviewer.)*

## Introduction:

This data project analyzes estimated real estate prices by answering vital questions from potential stakeholders to make data-driven decisions. The [dataset](https://www.kaggle.com/shwetabh123/mall-customers/activity) contains two decades' worth of data on median estimated residential values for single-family residences by zip code. Inside this project, I established vital questions to assist stakeholders in constructing potential strategies and informed business decisions for real estate investments.

## Methodology:

The main component of this project is to answer key questions to make data-driven decisions by analyzing estimated real estate value trends. To run my SQLite queries in jupyter notebook, I start by installing ipython-sql. Once installed, I load the sql module, then connected it to an existing database (home_value_trends.db) from my machine.

In [1]:
!pip install ipython-sql
import sqlalchemy
sqlalchemy.create_engine('sqlite:///home_value_trends.db')
%load_ext sql
%sql sqlite:///home_value_trends.db
print("Imports Successfully Installed")

Imports Successfully Installed


### 1. Table Info:

Before I begin the analysis, I first familiarize myself with the dataset.

In [2]:
%%sql
pragma table_info(home_value_data)

 * sqlite:///home_value_trends.db
Done.


[(0, 'zip_code', 'INTEGER', 0, None, 0),
 (1, 'city', 'TEXT', 0, None, 0),
 (2, 'state', 'TEXT', 0, None, 0),
 (3, 'metro', 'TEXT', 0, None, 0),
 (4, 'county', 'TEXT', 0, None, 0),
 (5, 'date', 'TEXT', 0, None, 0),
 (6, 'value', 'INTEGER', 0, None, 0)]

### 2. Glimpse of the dataset

In [3]:
%%sql

SELECT * 
FROM home_value_data
LIMIT 10;

 * sqlite:///home_value_trends.db
Done.


[(60657, 'Chicago', 'IL', 'Chicago-Naperville-Elgin', 'Cook County', '1996-04', 334200),
 (60657, 'Chicago', 'IL', 'Chicago-Naperville-Elgin', 'Cook County', '1996-05', 335400),
 (60657, 'Chicago', 'IL', 'Chicago-Naperville-Elgin', 'Cook County', '1996-06', 336500),
 (60657, 'Chicago', 'IL', 'Chicago-Naperville-Elgin', 'Cook County', '1996-07', 337600),
 (60657, 'Chicago', 'IL', 'Chicago-Naperville-Elgin', 'Cook County', '1996-08', 338500),
 (60657, 'Chicago', 'IL', 'Chicago-Naperville-Elgin', 'Cook County', '1996-09', 339500),
 (60657, 'Chicago', 'IL', 'Chicago-Naperville-Elgin', 'Cook County', '1996-10', 340400),
 (60657, 'Chicago', 'IL', 'Chicago-Naperville-Elgin', 'Cook County', '1996-11', 341300),
 (60657, 'Chicago', 'IL', 'Chicago-Naperville-Elgin', 'Cook County', '1996-12', 342600),
 (60657, 'Chicago', 'IL', 'Chicago-Naperville-Elgin', 'Cook County', '1997-01', 344400)]

### 3. How many distinct zip codes are in this dataset?

In [4]:
%%sql

SELECT COUNT(DISTINCT zip_code)AS distinct_zip
FROM home_value_data;

 * sqlite:///home_value_trends.db
Done.


[(15452,)]

### 4. How many zip codes are from each state?

In [5]:
%%sql

SELECT 
state,
COUNT(DISTINCT zip_code) as Count_Zip

FROM home_value_data

GROUP BY state
ORDER BY Count_Zip DESC;

 * sqlite:///home_value_trends.db
Done.


[('CA', 1230),
 ('NY', 1081),
 ('TX', 940),
 ('PA', 875),
 ('FL', 795),
 ('OH', 630),
 ('IL', 571),
 ('MI', 547),
 ('NJ', 501),
 ('NC', 489),
 ('TN', 436),
 ('VA', 428),
 ('WI', 426),
 ('IN', 420),
 ('MA', 419),
 ('MN', 416),
 ('GA', 375),
 ('WA', 348),
 ('MO', 326),
 ('MD', 322),
 ('CO', 261),
 ('KS', 242),
 ('AZ', 233),
 ('OK', 230),
 ('OR', 224),
 ('IA', 223),
 ('AL', 221),
 ('SC', 215),
 ('NH', 198),
 ('LA', 195),
 ('KY', 163),
 ('MS', 146),
 ('CT', 122),
 ('UT', 121),
 ('AR', 119),
 ('ID', 111),
 ('NV', 101),
 ('ME', 95),
 ('NE', 89),
 ('WV', 82),
 ('MT', 72),
 ('VT', 62),
 ('HI', 62),
 ('RI', 60),
 ('NM', 60),
 ('DE', 39),
 ('ND', 33),
 ('WY', 32),
 ('AK', 28),
 ('SD', 20),
 ('DC', 18)]

### 5. What is the range of years within this dataset?

- The range of years within this dataset is from 1996 - 2018

In [6]:
%%sql

SELECT DISTINCT substr(date, 1, 4) Years

FROM home_value_data;

 * sqlite:///home_value_trends.db
Done.


[('1996',),
 ('1997',),
 ('1998',),
 ('1999',),
 ('2000',),
 ('2001',),
 ('2002',),
 ('2003',),
 ('2004',),
 ('2005',),
 ('2006',),
 ('2007',),
 ('2008',),
 ('2009',),
 ('2010',),
 ('2011',),
 ('2012',),
 ('2013',),
 ('2014',),
 ('2015',),
 ('2016',),
 ('2017',),
 ('2018',)]

### 6. What is the range of estimated real estate prices across all states, using the most recent month of data available?

In the dataset, the most recent month of data available is November 2018.

In [7]:
%%sql

SELECT 
substr(date , 6) as month,
substr(date, 1,4) as year,
value

FROM home_value_data
WHERE month = '11' AND year = '2018'
LIMIT 10;

 * sqlite:///home_value_trends.db
Done.


[('11', '2018', 1053400),
 ('11', '2018', 336000),
 ('11', '2018', 1324200),
 ('11', '2018', 126600),
 ('11', '2018', 180200),
 ('11', '2018', 172000),
 ('11', '2018', 440000),
 ('11', '2018', 745300),
 ('11', '2018', 1120400),
 ('11', '2018', 4039900)]

### 7. In 2018, what was the average real estate price in each state?

In [8]:
%%sql

SELECT 
substr(date, 1 ,4) as year,
state,
round(avg(value) , 2) as average_price
FROM home_value_data
WHERE year = '2018'
GROUP BY state
;

 * sqlite:///home_value_trends.db
Done.


[('2018', 'AK', 323130.19),
 ('2018', 'AL', 148018.59),
 ('2018', 'AR', 135598.4),
 ('2018', 'AZ', 272885.1),
 ('2018', 'CA', 740019.65),
 ('2018', 'CO', 426110.55),
 ('2018', 'CT', 290724.29),
 ('2018', 'DC', 810175.76),
 ('2018', 'DE', 260174.83),
 ('2018', 'FL', 291752.21),
 ('2018', 'GA', 199381.24),
 ('2018', 'HI', 696020.23),
 ('2018', 'IA', 164192.58),
 ('2018', 'ID', 245507.13),
 ('2018', 'IL', 208972.47),
 ('2018', 'IN', 136027.47),
 ('2018', 'KS', 143034.07),
 ('2018', 'KY', 152607.7),
 ('2018', 'LA', 146550.02),
 ('2018', 'MA', 465974.59),
 ('2018', 'MD', 328104.01),
 ('2018', 'ME', 257127.94),
 ('2018', 'MI', 166879.66),
 ('2018', 'MN', 220615.76),
 ('2018', 'MO', 164175.46),
 ('2018', 'MS', 119996.01),
 ('2018', 'MT', 238423.11),
 ('2018', 'NC', 184655.27),
 ('2018', 'ND', 185347.66),
 ('2018', 'NE', 178589.07),
 ('2018', 'NH', 270041.32),
 ('2018', 'NJ', 397220.25),
 ('2018', 'NM', 226403.94),
 ('2018', 'NV', 333730.69),
 ('2018', 'NY', 374716.19),
 ('2018', 'OH', 147618.2),
 ('2018', 'OK', 118637.98),
 ('2018', 'OR', 327165.5),
 ('2018', 'PA', 183820.71),
 ('2018', 'RI', 316106.82),
 ('2018', 'SC', 190153.78),
 ('2018', 'SD', 198703.18),
 ('2018', 'TN', 149650.21),
 ('2018', 'TX', 204378.33),
 ('2018', 'UT', 326142.52),
 ('2018', 'VA', 280569.82),
 ('2018', 'VT', 224345.89),
 ('2018', 'WA', 404123.3),
 ('2018', 'WI', 187393.34),
 ('2018', 'WV', 122273.84),
 ('2018', 'WY', 233118.47)]

In [9]:
%%HTML
<div class='tableauPlaceholder' id='viz1627595848889' style='position: relative'><noscript><a href='#'><img alt='Average real estate price in 2018 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Av&#47;Averagerealestatepricein2018&#47;3_6&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='Averagerealestatepricein2018&#47;3_6' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Av&#47;Averagerealestatepricein2018&#47;3_6&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1627595848889');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Figure 1. The average home values in 2018. Visualization was developed and uploaded from Tableau Public by the user.

### 8. What is the average real estate price for each year within the dataset?

In [10]:
%%sql

SELECT 
substr(date, 1 ,4) as year,
round(avg(value),2) as average_price
FROM home_value_data
GROUP BY year;

 * sqlite:///home_value_trends.db
Done.


[('1996', 116572.12),
 ('1997', 120013.8),
 ('1998', 126946.38),
 ('1999', 136368.12),
 ('2000', 148941.86),
 ('2001', 161897.04),
 ('2002', 175223.95),
 ('2003', 191527.21),
 ('2004', 215468.95),
 ('2005', 243489.77),
 ('2006', 259340.59),
 ('2007', 256373.05),
 ('2008', 238012.42),
 ('2009', 218793.92),
 ('2010', 210078.66),
 ('2011', 199288.7),
 ('2012', 198771.48),
 ('2013', 211058.19),
 ('2014', 223840.72),
 ('2015', 236203.69),
 ('2016', 250059.9),
 ('2017', 266514.04),
 ('2018', 283594.63)]

In [11]:
%%HTML
<div class='tableauPlaceholder' id='viz1627596521786' style='position: relative'><noscript><a href='#'><img alt='Average estimated real estate price per year ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Av&#47;Averageestimatedrealestatepriceperyear&#47;3_7&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='Averageestimatedrealestatepriceperyear&#47;3_7' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Av&#47;Averageestimatedrealestatepriceperyear&#47;3_7&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1627596521786');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Figure 2. The average home values from 1996-2018. Visualization was developed and uploaded from Tableau Public by the user.

### 9. Which states have the highest average real estate price in November 2018?

In [12]:
%%sql

SELECT
substr(date,6) as month,
substr(date, 1,4) as year,
state,
round(avg(value),2) as average_price

FROM home_value_data
WHERE month = '11' AND year = '2018'
GROUP BY state
ORDER BY average_price DESC
LIMIT 1

;


 * sqlite:///home_value_trends.db
Done.


[('11', '2018', 'DC', 826572.22)]

### 10. Which states have the lowest average real estate price in November 2018?

In [13]:
%%sql

SELECT
substr(date,6) as month,
substr(date, 1,4) as year,
state,
round(avg(value),2) as average_price

FROM home_value_data
WHERE month = '11' AND year = '2018'
GROUP BY state
ORDER BY average_price ASC
LIMIT 1

;

 * sqlite:///home_value_trends.db
Done.


[('11', '2018', 'OK', 121545.65)]

In [14]:
%%HTML
<div class='tableauPlaceholder' id='viz1627596565649' style='position: relative'><noscript><a href='#'><img alt='Average real estate prices in November 2018 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Av&#47;AveragerealestatepricesinNovember2018&#47;3_8-3_9&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='AveragerealestatepricesinNovember2018&#47;3_8-3_9' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Av&#47;AveragerealestatepricesinNovember2018&#47;3_8-3_9&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1627596565649');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Figure 3. The average home values using the most recent month of data available. The highest real estate price is D.C with $ 826,572.22, and the lowest is O.K. with 121,545.65. Visualization was developed and uploaded from Tableau Public by the user.

### 11. In 2017, 2007, and 1997, which states have the highest and lowest real estate values?

In [15]:
%%sql
-- 2017 - MAX
SELECT
substr(date, 1 ,4) as year,
state,
round(avg(value),2) as average_price

FROM home_value_data
WHERE year = '2017' 
GROUP BY state
ORDER BY average_price DESC
LIMIT 1
;

 * sqlite:///home_value_trends.db
Done.


[('2017', 'DC', 778756.02)]

In [16]:
%%sql
-- 2017 - MIN
SELECT
substr(date, 1 ,4) as year,
state,
round(avg(value),2) as average_price

FROM home_value_data
WHERE year = '2017' 
GROUP BY state
ORDER BY average_price ASC
LIMIT 1
;

 * sqlite:///home_value_trends.db
Done.


[('2017', 'OK', 112451.7)]

In [17]:
%%sql
-- 2007 - MAX
SELECT
substr(date, 1 ,4) as year,
state,
round(avg(value),2) as average_price

FROM home_value_data
WHERE year = '2007' 
GROUP BY state
ORDER BY average_price DESC
LIMIT 1
;


 * sqlite:///home_value_trends.db
Done.


[('2007', 'HI', 611893.93)]

In [18]:
%%sql
-- 2017 - MIN
SELECT
substr(date, 1 ,4) as year,
state,
round(avg(value),2) as average_price

FROM home_value_data
WHERE year = '2007' 
GROUP BY state
ORDER BY average_price ASC
LIMIT 1
;

 * sqlite:///home_value_trends.db
Done.


[('2007', 'OK', 91773.39)]

In [19]:
%%sql
-- 1997 - MAX
SELECT
substr(date, 1 ,4) as year,
state,
round(avg(value),2) as average_price

FROM home_value_data
WHERE year = '1997' AND value IS NOT NULL
GROUP BY state
ORDER BY average_price DESC
LIMIT 1;

 * sqlite:///home_value_trends.db
Done.


[('1997', 'HI', 216377.12)]

In [20]:
%%sql
-- 1997 - MIN
SELECT
substr(date, 1 ,4) as year,
state,
round(avg(value),2) as average_price

FROM home_value_data
WHERE year = '1997' AND 
value IS NOT NULL
GROUP BY state
ORDER BY average_price ASC
LIMIT 1;

 * sqlite:///home_value_trends.db
Done.


[('1997', 'OK', 56325.62)]

In [21]:
%%HTML
<div class='tableauPlaceholder' id='viz1627596670078' style='position: relative'><noscript><a href='#'><img alt='Dashboard 2 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Av&#47;AveragePriceforeachstate-Geomap&#47;Dashboard2&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='AveragePriceforeachstate-Geomap&#47;Dashboard2' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Av&#47;AveragePriceforeachstate-Geomap&#47;Dashboard2&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1627596670078');                    var vizElement = divElement.getElementsByTagName('object')[0];                    if ( divElement.offsetWidth > 800 ) { vizElement.style.minWidth='420px';vizElement.style.maxWidth='650px';vizElement.style.width='100%';vizElement.style.minHeight='587px';vizElement.style.maxHeight='887px';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';} else if ( divElement.offsetWidth > 500 ) { vizElement.style.minWidth='420px';vizElement.style.maxWidth='650px';vizElement.style.width='100%';vizElement.style.minHeight='587px';vizElement.style.maxHeight='887px';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';} else { vizElement.style.width='100%';vizElement.style.height='727px';}                     var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Figure 4. Variations of average home prices in different states. Visualization was developed and uploaded from Tableau Public by the user.

In [22]:
%%HTML
<div class='tableauPlaceholder' id='viz1627596725364' style='position: relative'><noscript><a href='#'><img alt='Dashboard 1 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;To&#47;TopBottom10StatesbyHighestLowestAveragePrice&#47;Dashboard1&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='TopBottom10StatesbyHighestLowestAveragePrice&#47;Dashboard1' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;To&#47;TopBottom10StatesbyHighestLowestAveragePrice&#47;Dashboard1&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1627596725364');                    var vizElement = divElement.getElementsByTagName('object')[0];                    if ( divElement.offsetWidth > 800 ) { vizElement.style.minWidth='420px';vizElement.style.maxWidth='650px';vizElement.style.width='100%';vizElement.style.minHeight='587px';vizElement.style.maxHeight='887px';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';} else if ( divElement.offsetWidth > 500 ) { vizElement.style.minWidth='420px';vizElement.style.maxWidth='650px';vizElement.style.width='100%';vizElement.style.minHeight='587px';vizElement.style.maxHeight='887px';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';} else { vizElement.style.width='100%';vizElement.style.height='727px';}                     var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Figure 5 & 6. Top/Bottom 10 of the highest/lowest average home values from years 1997 to 2017. Visualization was developed and uploaded from Tableau Public by the user.

In [23]:
%%HTML
<div class='tableauPlaceholder' id='viz1627596909041' style='position: relative'><noscript><a href='#'><img alt='Ten Year Real Estate Average Value TrendFrom 2007 - 2017 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Te&#47;TenYearRealEstateAverageValueTrend&#47;3_11-LineChart&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='TenYearRealEstateAverageValueTrend&#47;3_11-LineChart' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Te&#47;TenYearRealEstateAverageValueTrend&#47;3_11-LineChart&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1627596909041');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Figure 7 Average home value trends from 2007 to 2017. Visualization was developed and uploaded from Tableau Public by the user.

### 12. What is the percent change in average real estate prices from 1997 to 2017 by state?

In [24]:
%%sql
WITH base_year AS (
SELECT
substr(date, 1 ,4) as start_year,
state,
round(avg(value),2) as average_price
FROM home_value_data
WHERE start_year  = '1997'
GROUP BY state),
end_year AS (
SELECT
substr(date, 1 ,4) as end_year,
state,
round(avg(value),2) as average_price
FROM home_value_data
WHERE end_year  = '2017'
GROUP BY state)

SELECT 
b.start_year,
b.state,
b.average_price,
e.end_year,
e.state,
e.average_price,
round(100.0 * (e.average_price - b.average_price) / b.average_price , 2)as percent_change
FROM base_year as b 
LEFT JOIN end_year as e
ON b.state = e.state
;

 * sqlite:///home_value_trends.db
Done.


[('1997', 'AK', 134421.13, '2017', 'AK', 318021.43, 136.59),
 ('1997', 'AL', 89567.68, '2017', 'AL', 139148.3, 55.36),
 ('1997', 'AR', 72332.22, '2017', 'AR', 128088.31, 77.08),
 ('1997', 'AZ', 122682.65, '2017', 'AZ', 255581.97, 108.33),
 ('1997', 'CA', 207479.51, '2017', 'CA', 688609.97, 231.89),
 ('1997', 'CO', 154106.44, '2017', 'CO', 394511.46, 156.0),
 ('1997', 'CT', 159184.02, '2017', 'CT', 280047.68, 75.93),
 ('1997', 'DC', 189769.44, '2017', 'DC', 778756.02, 310.37),
 ('1997', 'DE', 126937.0, '2017', 'DE', 248462.39, 95.74),
 ('1997', 'FL', 108010.5, '2017', 'FL', 272582.98, 152.37),
 ('1997', 'GA', 98839.32, '2017', 'GA', 184896.8, 87.07),
 ('1997', 'HI', 216377.12, '2017', 'HI', 653451.75, 202.0),
 ('1997', 'IA', 76168.14, '2017', 'IA', 156758.26, 105.81),
 ('1997', 'ID', 102877.86, '2017', 'ID', 223306.91, 117.06),
 ('1997', 'IL', 123320.01, '2017', 'IL', 201485.96, 63.38),
 ('1997', 'IN', 87203.9, '2017', 'IN', 125044.21, 43.39),
 ('1997', 'KS', 88037.43, '2017', 'KS', 136451.0, 54.99),
 ('1997', 'KY', 82019.24, '2017', 'KY', 143375.72, 74.81),
 ('1997', 'LA', 69803.65, '2017', 'LA', 142874.7, 104.68),
 ('1997', 'MA', 168343.61, '2017', 'MA', 438244.63, 160.33),
 ('1997', 'MD', 144904.11, '2017', 'MD', 314660.25, 117.15),
 ('1997', 'ME', 96941.37, '2017', 'ME', 240862.11, 148.46),
 ('1997', 'MI', 99656.75, '2017', 'MI', 155352.13, 55.89),
 ('1997', 'MN', 91002.4, '2017', 'MN', 207513.66, 128.03),
 ('1997', 'MO', 97086.51, '2017', 'MO', 153369.2, 57.97),
 ('1997', 'MS', 86224.88, '2017', 'MS', 115630.65, 34.1),
 ('1997', 'MT', 157241.5, '2017', 'MT', 221810.42, 41.06),
 ('1997', 'NC', 91369.82, '2017', 'NC', 174062.83, 90.5),
 ('1997', 'ND', None, '2017', 'ND', 183148.99, None),
 ('1997', 'NE', 93304.28, '2017', 'NE', 166257.87, 78.19),
 ('1997', 'NH', 109703.39, '2017', 'NH', 252847.1, 130.48),
 ('1997', 'NJ', 175576.6, '2017', 'NJ', 375706.84, 113.98),
 ('1997', 'NM', 139284.75, '2017', 'NM', 212525.42, 52.58),
 ('1997', 'NV', 157095.79, '2017', 'NV', 301538.53, 91.95),
 ('1997', 'NY', 126348.63, '2017', 'NY', 359342.49, 184.41),
 ('1997', 'OH', 92003.07, '2017', 'OH', 138364.51, 50.39),
 ('1997', 'OK', 56325.62, '2017', 'OK', 112451.7, 99.65),
 ('1997', 'OR', 132921.59, '2017', 'OR', 306799.07, 130.81),
 ('1997', 'PA', 89577.67, '2017', 'PA', 174637.41, 94.96),
 ('1997', 'RI', 126419.31, '2017', 'RI', 294574.86, 133.01),
 ('1997', 'SC', 89122.47, '2017', 'SC', 176363.72, 97.89),
 ('1997', 'SD', 59295.0, '2017', 'SD', 184605.42, 211.33),
 ('1997', 'TN', 68213.77, '2017', 'TN', 138124.43, 102.49),
 ('1997', 'TX', 113221.74, '2017', 'TX', 192054.76, 69.63),
 ('1997', 'UT', 162484.05, '2017', 'UT', 294948.76, 81.52),
 ('1997', 'VA', 122760.53, '2017', 'VA', 269285.32, 119.36),
 ('1997', 'VT', 99300.0, '2017', 'VT', 219629.97, 121.18),
 ('1997', 'WA', 147066.59, '2017', 'WA', 369567.15, 151.29),
 ('1997', 'WI', 111299.37, '2017', 'WI', 176782.43, 58.84),
 ('1997', 'WV', 69983.47, '2017', 'WV', 118281.61, 69.01),
 ('1997', 'WY', 135059.57, '2017', 'WY', 222159.9, 64.49)]

### 13. What is the percent change in average real estate prices from 2007 to 2017 by state?

In [25]:
%%sql
WITH base_year AS (
SELECT
substr(date, 1 ,4) as start_year,
state,
round(avg(value),2) as average_price
FROM home_value_data
WHERE start_year  = '2007'
GROUP BY state),
end_year AS (
SELECT
substr(date, 1 ,4) as end_year,
state,
round(avg(value),2) as average_price
FROM home_value_data
WHERE end_year  = '2017'
GROUP BY state)

SELECT 
b.start_year,
b.state,
b.average_price,
e.end_year,
e.state,
e.average_price,
round(100.0 * (e.average_price - b.average_price) / b.average_price , 2)as percent_change
FROM base_year as b 
LEFT JOIN end_year as e
ON b.state = e.state
;

 * sqlite:///home_value_trends.db
Done.


[('2007', 'AK', 275928.87, '2017', 'AK', 318021.43, 15.25),
 ('2007', 'AL', 148621.43, '2017', 'AL', 139148.3, -6.37),
 ('2007', 'AR', 120089.75, '2017', 'AR', 128088.31, 6.66),
 ('2007', 'AZ', 295021.39, '2017', 'AZ', 255581.97, -13.37),
 ('2007', 'CA', 598141.04, '2017', 'CA', 688609.97, 15.13),
 ('2007', 'CO', 313130.9, '2017', 'CO', 394511.46, 25.99),
 ('2007', 'CT', 341385.11, '2017', 'CT', 280047.68, -17.97),
 ('2007', 'DC', 599893.98, '2017', 'DC', 778756.02, 29.82),
 ('2007', 'DE', 304398.15, '2017', 'DE', 248462.39, -18.38),
 ('2007', 'FL', 299784.12, '2017', 'FL', 272582.98, -9.07),
 ('2007', 'GA', 182825.09, '2017', 'GA', 184896.8, 1.13),
 ('2007', 'HI', 611893.93, '2017', 'HI', 653451.75, 6.79),
 ('2007', 'IA', 128333.03, '2017', 'IA', 156758.26, 22.15),
 ('2007', 'ID', 210176.45, '2017', 'ID', 223306.91, 6.25),
 ('2007', 'IL', 230522.74, '2017', 'IL', 201485.96, -12.6),
 ('2007', 'IN', 120574.49, '2017', 'IN', 125044.21, 3.71),
 ('2007', 'KS', 144679.71, '2017', 'KS', 136451.0, -5.69),
 ('2007', 'KY', 126206.01, '2017', 'KY', 143375.72, 13.6),
 ('2007', 'LA', 129153.11, '2017', 'LA', 142874.7, 10.62),
 ('2007', 'MA', 377822.66, '2017', 'MA', 438244.63, 15.99),
 ('2007', 'MD', 369100.47, '2017', 'MD', 314660.25, -14.75),
 ('2007', 'ME', 231709.8, '2017', 'ME', 240862.11, 3.95),
 ('2007', 'MI', 146659.44, '2017', 'MI', 155352.13, 5.93),
 ('2007', 'MN', 199287.95, '2017', 'MN', 207513.66, 4.13),
 ('2007', 'MO', 151943.81, '2017', 'MO', 153369.2, 0.94),
 ('2007', 'MS', 131843.64, '2017', 'MS', 115630.65, -12.3),
 ('2007', 'MT', 215423.5, '2017', 'MT', 221810.42, 2.96),
 ('2007', 'NC', 170933.84, '2017', 'NC', 174062.83, 1.83),
 ('2007', 'ND', 124564.08, '2017', 'ND', 183148.99, 47.03),
 ('2007', 'NE', 137831.83, '2017', 'NE', 166257.87, 20.62),
 ('2007', 'NH', 254791.45, '2017', 'NH', 252847.1, -0.76),
 ('2007', 'NJ', 440055.9, '2017', 'NJ', 375706.84, -14.62),
 ('2007', 'NM', 232761.79, '2017', 'NM', 212525.42, -8.69),
 ('2007', 'NV', 354913.26, '2017', 'NV', 301538.53, -15.04),
 ('2007', 'NY', 318182.44, '2017', 'NY', 359342.49, 12.94),
 ('2007', 'OH', 131262.94, '2017', 'OH', 138364.51, 5.41),
 ('2007', 'OK', 91773.39, '2017', 'OK', 112451.7, 22.53),
 ('2007', 'OR', 275055.33, '2017', 'OR', 306799.07, 11.54),
 ('2007', 'PA', 176372.29, '2017', 'PA', 174637.41, -0.98),
 ('2007', 'RI', 322582.22, '2017', 'RI', 294574.86, -8.68),
 ('2007', 'SC', 180254.11, '2017', 'SC', 176363.72, -2.16),
 ('2007', 'SD', 143045.61, '2017', 'SD', 184605.42, 29.05),
 ('2007', 'TN', 117068.67, '2017', 'TN', 138124.43, 17.99),
 ('2007', 'TX', 152363.82, '2017', 'TX', 192054.76, 26.05),
 ('2007', 'UT', 254423.06, '2017', 'UT', 294948.76, 15.93),
 ('2007', 'VA', 283551.94, '2017', 'VA', 269285.32, -5.03),
 ('2007', 'VT', 224758.48, '2017', 'VT', 219629.97, -2.28),
 ('2007', 'WA', 316000.91, '2017', 'WA', 369567.15, 16.95),
 ('2007', 'WI', 169244.15, '2017', 'WI', 176782.43, 4.45),
 ('2007', 'WV', 112406.01, '2017', 'WV', 118281.61, 5.23),
 ('2007', 'WY', 202277.08, '2017', 'WY', 222159.9, 9.83)]

### Results:
The trend in home values for each state from 1997 to 2017 has a positive percent change. However, between 2007 and 2017, various homes lost their value. The bottom five states with the most negative percent change are D.E. -18.4%, C.T. -18%, N.V. -15%, M.D. -14.8%, and N.J. -14.6%. In contrast, N.D., D.C., S.D., T.X., and C.O were the top five states, between 2007 and 2017, with the highest percent change. 

Overall the top five states with the highest percent change from 1997 to 2017 are D.C. +310.4%, C.A.+231.9%, S.D. +211.3% , H.I. +202%, and N.Y +184.4%.